include/seqan3/io/alignment_file/format_sam_base.hpp

```diff
@@ -447,165 +447,175 @@ inline void format_sam_base::read_header(stream_view_type && stream_view,
     {
-        std::ranges::next(std::ranges::begin(stream_view)); // skip @
+        ++it;                                                           // skip @
 
-        if (is_char<'H'>(*std::ranges::begin(stream_view))) // HD (header) tag
+        switch (*it)
         {
-            parse_tag_value(hdr.format_version); // parse required VN (version) tag
-
-            // The SO, SS and GO tag are optional and can appear in any order
-            while (is_char<'\t'>(*std::ranges::begin(stream_view)))
+            case 'H': /* HD (header) tag */
             {
-                std::ranges::next(std::ranges::begin(stream_view)); // skip tab
-                std::string * who = std::addressof(hdr.grouping);
+                parse_tag_value(hdr.format_version);                    // parse required VN (version) tag
 
-                if (is_char<'S'>(*std::ranges::begin(stream_view)))
+                // The SO, SS and GO tag are optional and can appear in any order
+                while (is_char<'\t'>(*it))
                 {
-                    std::ranges::next(std::ranges::begin(stream_view)); // skip S
-
-                    if (is_char<'O'>(*std::ranges::begin(stream_view))) // SO (sorting) tag
-                        who = std::addressof(hdr.sorting);
-                    else if (is_char<'S'>(*std::ranges::begin(stream_view))) // SS (sub-order) tag
-                        who = std::addressof(hdr.subsorting);
-                    else
-                        throw format_error{std::string{"Illegal SAM header tag: S"} +
-                                           std::string{static_cast<char>(*std::ranges::begin(stream_view))}};
+                    ++it;                                               // skip tab
+                    std::string * who = std::addressof(hdr.grouping);
+
+                    if (is_char<'S'>(*it))
+                    {
+                        ++it;                                           // skip S
+
+                        if (is_char<'O'>(*it))                          // SO (sorting) tag
+                            who = std::addressof(hdr.sorting);
+                        else if (is_char<'S'>(*it))                     // SS (sub-order) tag
+                            who = std::addressof(hdr.subsorting);
+                        else
+                            throw format_error{std::string{"Illegal SAM header tag: S"} + *it};
+                    }
+                    else if (!is_char<'G'>(*it))                        // GO (grouping) tag
+                    {
+                        throw format_error{std::string{"Illegal SAM header tag in @HG starting with: "} + *it};
+                    }
+
+                    parse_tag_value(*who);
                 }
-                else if (!is_char<'G'>(*std::ranges::begin(stream_view))) // GO (grouping) tag
+                ++it;                                                   // skip newline
+            } break;
```

Comment:
This `break` still belongs to the `case`?
If so, move the `break` into the scope that just ended, to make that clear?

(Same for other cases)

Reply:
Hm, I always put it behind the block to indicate that the `break` belongs to the `case` (which is also outside the `{ ... } `). And having the break directly before the subsequent `case` clearly indicates that there is no fallthrough.

But it's really not that important to me :)

Reply:
That's fair, I was just really confused why you end the scope and then put the `break`